### PR TITLE
Register Microsoft.Monitor RP in Network cmdlets

### PR DESCRIPTION
**Pull request by GokulPrasad-Work**
## Description

Adding Microsoft.Monitor RP in Network PrivateLinkService cmdlet

Microsoft.Monitor RP supports
 - List Private Link Resource API
 - Get Private Endpoint Connection API
 - List Private Endpoint Connection API

It doesn't support Get Private Link Resource API yet

## Checklist

- [*] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Network/Network/ChangeLog.md [hunk 1]
```diff
@@ -24,6 +24,7 @@
     - `New-AzFirewallPolicy`
     - `Set-AzFirewallPolicy`
 * Fixed a bug that reverts classic fw private ranges to default when doing get & set
+* Onboard Microsoft.Monitor to Private Link Common Cmdlets
```
Comment (BethanyZhou) [inline src/Network/Network/ChangeLog.md:27]:
```suggestion
* Onboarded `Microsoft.Monitor/accounts` to Private Link Cmdlets
```

**Review by BethanyZhou — APPROVED** [review]
Looks good to me

**Comment by GokulPrasad-Work** [comment]
Hello, Is there a way for us to take the intermediate build and install it in local machine to validate the new changes?

**Comment by BethanyZhou** [comment]
Yes, we do have.
- If you just want to test during the implementation, try https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/azure-powershell-developer-guide.md#enable-running-powershell-when-debugging
- If you want to install new build in local, please follow this guide to [Create unsigned nuget packages locally](https://eng.ms/docs/cloud-ai-platform/azure-core/azure-management-and-platforms/control-plane-bburns/azure-cli-tools-azure-cli-powershell-and-terraform/azure-cli-tools/devguide/azps/implementation#create-unsigned-nuget-packages-locally).

**Comment by GokulPrasad-Work** [comment]
Hello @BethanyZhou, After the release, I tried to execute the Approve-AzPrivateEndpointConnection -ResourceId <PrivateEndpointConnectionResourceId>, but I am getting the below error:
Approve-AzPrivateEndpointConnection: Value cannot be null. (Parameter 'Id')
StatusCode: 400
ReasonPhrase: Bad Request
ErrorCode: BadArgument
ErrorMessage: Value cannot be null. (Parameter 'Id')
OperationID : 17228718-d310-4a95-bceb-cc20f19d0aba

Looks like the request is not populated with the id field  which is given as "PrivateEndpointConnectionResourceId" in the cmdlet input.
Expected request payload is like below:
{ 
     "id": "/subscriptions/{subscriptionId}/resourceGroups/{groupName}/providers/Microsoft.[Service]/{resourceType}/{resourceName}/privateEndpointConnections/{privateEndpointConnectionName}",
"properties": 
     { "privateLinkServiceConnectionState": 
             { 
               "status": "Approved", 
               "description": "Approved by johndoe@company.com" 
             }
      }
 }

Can you please help in debugging this issue?

**Comment by BethanyZhou** [comment]
@GokulPrasad-Work , please provide the command and its parameters you used when hit this issue to allow me reproduce this issue.

**Comment by GokulPrasad-Work** [comment]
> @GokulPrasad-Work , please provide the command and its parameters you used when hit this issue to allow me reproduce this issue.

@BethanyZhou 
Please see the below snapshot where the Approve private endpoint connection command is failing. Service expects Id parameter to be populated in the request. Looks like it is not getting populated in Powershell module which is causing this bad request failure from the service. Please feel free to ping me to discuss more on this issue. Thank you!

Command: AzPrivateEndpointConnection -ResourceId /subscriptions/<subId>/resourcegroups/<rgName>/providers/microsoft.monitor/accounts/<AccountName>/privateEndpointConnections/<peConnectionName>  -Description Approved
image

**Pull request merged**